scripts/abi_artifact_generator/templates/setup_arguments.S.jinja_template

```diff
@@ -0,0 +1,53 @@
+/*#-
+This template file is distributed under the MIT License. See LICENSE.md for details.
+#*/
+
+/*(- generation_notice )*/
+
+/*{ if current_style == 'gcc' -}*/
+#include "../abi.h"
+
+#if defined(TARGET_x86_64) || defined(TARGET_i386)
+.intel_syntax noprefix
+#endif
+/*{- elif current_style == 'msvc' -}*/
+ifndef rax
+.model flat, c
+endif
+
+extern randomized_state: /*( intel_register_type )*/
+extern printable_data: /*( intel_register_type )*/
+extern saved_return_address: /*( intel_register_type )*/
+
+/*{ for name, arguments in argument_functions.items() -}*/
+extern test_/*( name )*/: proc
+/*{ endfor }*/
+
+.code
+/*{- endif }*/
+
+/*{ for name, arguments in argument_functions.items() }*/
+/*{- if current_style == 'gcc' -}*/
+.global setup_/*( name )*/
+setup_/*( name )*/:
+/*{- elif current_style == 'msvc' -}*/
+setup_/*( name )*/ proc
+/*{- endif }*/
+  /*( save_return_address )*/
+  /*( fill_stack_with_random_data )*/
+  /*( fill_registers_with_random_data )*/
+  /*( call_a_function("test_" + name) )*/
```

Comment:
As far as I can see, `save_return_address`, `fill_stack_with_random_data`, `fill_registers_with_random_data` are used only once, here.

Can we merge them and call it like "setup_prologue"?

We're missing docs, but `save_return_address` does not convey very well that it will be called at the entry of the function.

More in general, can you look into how much can we squash together things that always go together? Like `restore_return_address` and `return_from_function`.

Reply:
Also, add a comment in the output so it's clear what's going on.

Something like

```patch
diff --git b/./abi_artifact_generated/x86_64/gcc_setup_arguments-original.S a/./abi_artifact_generated/x86_64/gcc_setup_arguments.S
index cce88c0..2cbeb67 100644
--- b/./abi_artifact_generated/x86_64/gcc_setup_arguments-original.S
+++ a/./abi_artifact_generated/x86_64/gcc_setup_arguments.S
@@ -10,8 +10,11 @@
 
 .global setup_single_8_bit_argument
 setup_single_8_bit_argument:
+  ; Save return address
   pop rax
   mov qword ptr [saved_return_address], rax
+
+  ; Populate the stack with random data
   mov rax, qword ptr [randomized_state + 616]
   push rax
   mov rax, qword ptr [randomized_state + 608]
```

Reply:
Done.